src/main/java/com/capstone/doctorsOffice/controllers/PrescriptionController.java

```diff
@@ -0,0 +1,48 @@
+package com.capstone.doctorsOffice.controllers;
+
+import com.capstone.doctorsOffice.dtos.PrescriptionDto;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+import com.capstone.doctorsOffice.services.PrescriptionService;
+import com.capstone.doctorsOffice.repositories.PrescriptionRepository;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("/api/prescriptions")
+
+
+public class PrescriptionController {
+    @Autowired
+    private PrescriptionService prescriptionService;
+    @Autowired
+    private PrescriptionRepository prescriptionRepository;
+
+    @GetMapping("")
+    public List<PrescriptionDto> getAllPrescription() {
+        return prescriptionService.getAllPrescriptions();
+    }
+
+    @GetMapping("/{id}")
+    public Optional<PrescriptionDto> getPrescriptionById(@PathVariable Long id) {
+        return prescriptionService.getPrescriptionById(id);
+    }
+
+    @PostMapping("/new-prescription")
+    public List<String> addNewPrescription(@RequestBody PrescriptionDto prescriptionDto) {
+        return prescriptionService.addNewPrescription(prescriptionDto);
+    }
+
```

Comment:
The `addNewPrescription()` in the Prescription service just return null.
change this to the `addPrescription` method from the Service class.